apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js

```diff
@@ -77,14 +77,35 @@ router.get(
         res.locals.assessment_instance.id,
         false,
       );
-
+      const fingerprintNumbers = {};
+      let i = 1;
       const stringifier = stringifyStream({
         header: true,
-        columns: ['Time', 'Auth user', 'Event', 'Instructor question', 'Student question', 'Data'],
+        columns: [
+          'Time',
+          'Auth user',
+          'Fingerprint',
+          'IP Address',
+          'Event',
+          'Instructor question',
+          'Student question',
+          'Data',
+        ],
         transform(record) {
+          if (record.client_fingerprint) {
+            if (!fingerprintNumbers[record.client_fingerprint.id]) {
+              fingerprintNumbers[record.client_fingerprint.id] = i;
+              i++;
+            }
+            record.client_fingerprint_number = fingerprintNumbers[record.client_fingerprint.id];
+          }
```

Comment:
This is the piece that is calculating the client fingerprint number and adding it to the row. Currently this is functional. However, I am not familiar with streaming transformers so if this is not the correct way to do so, please let me know.

Reply:
This looks correct to me!